### PR TITLE
Throw on duplicate `__proto__` props followed by assignment

### LINKED ISSUE #13930: [Bug]: Babel does not throw duplicated __proto__ when followed by a property whose value is an assignment expression
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

Programmatic API (`babel.transform`, `babel.parse`)

### Input code

```js
({
  __proto__: a,
  __proto__: a,
  a: a = 1
})
```

[REPL](https://babel.dev/repl#?browsers=firefox%2091&build=&builtIns=false&corejs=3.6&spec=false&loose=false&code_lz=BQbwUABBD60A4CcD2AXJsBcECGAaSM8yamO-U2W2EAvBAIxgC-AlEA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=stage-1&prettier=false&targets=&version=7.16.2&externalPlugins=&assumptions=%7B%7D)

### Configuration file name

_No response_

### Configuration

N. A.

### Current and expected behavior

Currently Babel parses it without errors.

It should throw 

```
Redefinition of __proto__ property.
```

just like Babel correctly throws in the following example:

```js
({
  __proto__: a,
  __proto__: a,
})
```

### Environment

REPL

### Possible solution

This issue is very similar to https://github.com/babel/babel/issues/10262. 

Like how we handle `shorthandAssign`

https://github.com/babel/babel/blob/7250d2562be881f9cdc24c6646468e0f3afcaf9d/packages/babel-parser/src/parser/expression.js#L317-L319

we should reset `doubleProto` only when it is used correctly.

https://github.com/babel/babel/blob/7250d2562be881f9cdc24c6646468e0f3afcaf9d/packages/babel-parser/src/parser/expression.js#L312

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by The-x-Theorist**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #13930 
| Patch: Bug Fix?          | Y
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

doubleProto will get reset only when it's used correctly, Added a check to verify the position of doubleProto if it's verified it gets reset.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/13951"></a>

### CODE REVIEW

File: packages/babel-parser/src/parser/expression.js [hunk 1]
```diff
@@ -309,7 +309,10 @@ export default class ExpressionParser extends LValParser {
 
       if (this.match(tt.eq)) {
         node.left = this.toAssignable(left, /* isLHS */ true);
-        refExpressionErrors.doubleProto = -1; // reset because double __proto__ is valid in assignment expression
+
+        if (refExpressionErrors.doubleProto >= node.left.start) {
```
Comment (JLHwung) [inline packages/babel-parser/src/parser/expression.js:313]: We can replace `node.left.start` by `startPos`, saving unnecessary double property access.

File: packages/babel-parser/test/fixtures/es2015/duplicate-proto/without-assignment-expression/input.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+({
+  __proto__: a,
+  __proto__: a,
+})
```
Comment (JLHwung) [inline packages/babel-parser/test/fixtures/es2015/duplicate-proto/without-assignment-expression/input.js:4]: We already have a similar test in `packages/babel-parser/test/fixtures/es2015/uncategorised/348/input.js`, can you remove that uncategorised tests?

File: packages/babel-parser/src/parser/expression.js (hunk 1 above)
Comment (The-x-Theorist) [reply packages/babel-parser/src/parser/expression.js:313]: Okay, I will do the same.

File: packages/babel-parser/test/fixtures/es2015/duplicate-proto/without-assignment-expression/input.js (hunk 2 above)
Comment (The-x-Theorist) [reply packages/babel-parser/test/fixtures/es2015/duplicate-proto/without-assignment-expression/input.js:4]: Okay, I will remove it.

**Comment by The-x-Theorist** [comment]
@JLHwung Does this commit looks good?

**Review by JLHwung — APPROVED** [review]
Thanks.

**Review by nicolo-ribaudo — COMMENTED** [review]
Thank you!

**Review by nicolo-ribaudo — APPROVED** [review]

**Pull request merged**